src/Engine/ProtoCore/Reflection/Mirror.cs

```diff
@@ -335,26 +335,27 @@ internal ClassMirror(StackValue svData, ProtoCore.Core core)
             }
 
             /// <summary>
-            /// Returns the constructors and static methods and properties 
-            /// belonging to the type and its base types
+            /// Returns a list of constructors and static methods and properties 
+            /// belonging to the type and its base types. Filters out repeating names.
             /// </summary>
             /// <returns></returns>
             public IEnumerable<StaticMirror> GetMembers()
             {
                 // TODO: Factor out reflection functionality for both LibraryServices and Mirrors
                 List<StaticMirror> members = new List<StaticMirror>();
+                members.AddRange(GetConstructors());
+                members.AddRange(GetFunctions().Where(m => m.IsStatic));
+                members.AddRange(GetProperties().Where(m => m.IsStatic));
 
-                IEnumerable<ClassMirror> baseClasses = this.GetClassHierarchy();
+                IEnumerable<ClassMirror> baseClasses = GetClassHierarchy();
                 foreach (var baseClass in baseClasses)
                 {
-                    members.AddRange(baseClass.GetFunctions().Where(m => m.IsStatic).GroupBy(x => x.Name).Select(y => y.First()));
-                    members.AddRange(baseClass.GetProperties().Where(m => m.IsStatic).GroupBy(x => x.Name).Select(y => y.First()));
+                    members.AddRange(baseClass.GetFunctions().Where(m => m.IsStatic));
+                    members.AddRange(baseClass.GetProperties().Where(m => m.IsStatic));
                 }
 
-                members.AddRange(this.GetConstructors().GroupBy(x => x.Name).Select(y => y.First()));
-                members.AddRange(this.GetFunctions().Where(m => m.IsStatic).GroupBy(x => x.Name).Select(y => y.First()));
-                members.AddRange(this.GetProperties().Where(m => m.IsStatic).GroupBy(x => x.Name).Select(y => y.First()));
-                return members;
+                // Return a list of members with unique names.
+                return members.GroupBy(x => x.Name).Select(x => x.First()).ToList();
```

Comment:
Seems like we were filtering out members with duplicate names within the scope of each class.
I moved the filtering logic at the end so that we remove duplicates across all class hierarchies .